src/test/app/Wasm_test.cpp

```diff
@@ -695,6 +695,1089 @@ struct Wasm_test : public beast::unit_test::suite
+                if (BEAST_EXPECT(re.has_value()))
+                {
+                    // Verify results are consistent
+                    int expected[] = {8, 13, 21, 34, 55};
+                    BEAST_EXPECTS(
+                        re->result == expected[i],
+                        std::to_string(re->result) +
+                            " != " + std::to_string(expected[i]));
+                }
+            }
+        }
+    }
+
+    void
+    testWamrEdgeCases()
+    {
+        testcase("WAMR edge cases and boundary conditions");
+
+        using namespace test::jtx;
+        Env env(*this);
+        auto& engine = WasmEngine::instance();
+
+        // Test empty WASM module
+        {
+            Bytes emptyModule = {
+                0x00, 0x61, 0x73, 0x6D, 0x01, 0x00, 0x00, 0x00};  // Just header
+            auto re = engine.run(emptyModule, "_start", {});
+            BEAST_EXPECT(!re.has_value());  // Should fail gracefully
+        }
+
+        // Test WASM module with no exports
+        {
+            Bytes noExportsModule = {
+                0x00, 0x61, 0x73, 0x6D, 0x01, 0x00, 0x00, 0x00,  // WASM header
+                0x01, 0x04, 0x01, 0x60, 0x00, 0x00,  // Type section: () -> ()
+                0x03, 0x02, 0x01, 0x00,              // Function section
+                0x0A, 0x04, 0x01, 0x02, 0x00, 0x0B   // Code section: empty
+                                                     // function
+                // No export section
+            };
+
+            auto re = engine.run(noExportsModule, "missing", {});
+            BEAST_EXPECT(!re.has_value());  // Should fail - no exports
+        }
+
+        // Test function with maximum valid parameters
+        {
+            auto const wasmStr = boost::algorithm::unhex(fibWasmHex);
+            Bytes const wasm(wasmStr.begin(), wasmStr.end());
+
+            // Test with maximum integer value as parameter
+            auto re = engine.run(wasm, "fib", wasmParams(INT32_MAX));
+            (void)re;  // Testing crash resistance with extreme values
+            // May fail due to computation limits or succeed with some result
+            // Don't enforce specific behavior, just ensure no crash
+        }
+
+        // Test gas limit edge cases
+        {
+            auto const wasmStr = boost::algorithm::unhex(fibWasmHex);
+            Bytes const wasm(wasmStr.begin(), wasmStr.end());
+
+            // Test with gas limit of 0
+            auto re1 = engine.run(wasm, "fib", wasmParams(1), {}, nullptr, 0);
+            BEAST_EXPECT(!re1.has_value());  // Should fail immediately
+
+            // Test with gas limit of 1
+            auto re2 = engine.run(wasm, "fib", wasmParams(1), {}, nullptr, 1);
+            BEAST_EXPECT(!re2.has_value());  // Should fail quickly
+
+            // Test with extremely large gas limit
+            auto re3 =
+                engine.run(wasm, "fib", wasmParams(5), {}, nullptr, LLONG_MAX);
```

Comment:
Also try with -1. Should be the same, as -1 and std::numeric_limit<int64_t>::max()  gas values have the same default logic.